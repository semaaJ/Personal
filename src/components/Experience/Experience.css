.root { 
    margin-top: 50px;
    justify-content: center;
    flex-grow: 1;
    background-color: white;
    display: flex;
    height: 224;
}

.tabContentContainer {
    text-align: left;
}

.title {
    font-weight: bold !important;
    font: 18px 'Inconsolata', monospace;
    color: #383737;
}

.tabs {
    height: 100px;
    border-right: 1px solid black;
}

.employerLink {
    text-decoration: none;
    font-weight: bold !important;
    font: 18px 'Inconsolata', monospace;
    color: #383737;
}

.jobDetails {
    padding-top: 4px;
    list-style: none;
    width: 750px;
}

.jobDetails li:before { 
    content: "";
    border-color: transparent #111;
    border-style: solid;
    border-width: 0.35em 0 0.35em 0.45em;
    display: block;
    height: 0;
    width: 0;
    left: -1em;
    top: 1em;
    position: relative;
}

