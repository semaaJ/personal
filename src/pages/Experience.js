import romac from '../static/romachomee.png';
import mellio from '../static/mellio.png';
import cavancbt from '../static/cavancbt.png';

const Experience = () => {
    return (
        <div className="d-f" style={{ marginLeft: "55px", marginRight: "55px" }}>
          <div className="d-f w-100 jc-sb">
            <div className="d-f fd-c">
              <h1>experience</h1>
              <div className="d-f fd-c">
                <div className="d-f fd-c mb-22" style={{ width: "550px", marginRight: "25px" }}>
                  <div className="d-f jc-sb al-c">
                    <h2><span className="primaryDark">mellio</span>, dublin</h2>
                    <h4>aug 20 - current</h4>
                  </div>
                  <h2>product manager & developer</h2>
                  <p className="mb-4 mt-0">participated in the development of an saas product which enables mechanical workshops to manage, monitor and reward employees</p>
                  <p className="mb-4 mt-0">managed stakeholder communication</p>
                  <p className="mb-4 mt-0">and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum</p>
                  <p className="mb-4 mt-0">participated in the development of an saas product which enables mechanical workshops to manage, monitor and reward employees</p>
                  <hr className="divide"/>
                </div>
  
                <div className="d-f fd-c" style={{ width: "550px", marginRight: "25px", marginBottom: "25px"}}>
                  <div className="d-f jc-sb al-c">
                    <h2><span className="primaryDark">shutterstock</span>, dublin</h2>
                    <h4>apr 19 - sept 19</h4>
                  </div>
                  <h2>software engineer intern</h2>
                  <p className="mb-4 mt-0">worked, using agile & scrum methods, alongside a team developing an application which would provide a critical connectivity channel between stakeholders.</p>
                  <p className="mb-4 mt-0">translated designs to high quality & reusable react components.</p>
                  <p className="mb-4 mt-0">collaborated with editors to gather feedback for iterative design</p>
                  <p className="mb-4 mt-0">deployed code which met company standards of 85% test coverage (via jest) & participated in peer code reviews for adherence to coding standards.</p>
                  <p className="mb-4 mt-0">experienced developing back end services (Typescript)</p>
                  <p className="mb-4 mt-0">participated in a company hackathon.</p>
                  <hr className="divide"/>
                </div>
  
                <div className="d-f fd-c" style={{ width: "550px", marginRight: "25px", marginBottom: "25px"}}>
                <div className="d-f jc-sb al-c">
                    <h2><span className="primaryDark">dublin city university</span>, dublin</h2>
                    <h4>jan 19 - feb 19</h4>
                  </div>
                  <h2>lab assistant</h2>
                  <h4>feb 2019 - mar 2019</h4>
                  <p className="m-0">worked, using agile & scrum methods, alongside a team developing an application which would provide a critical connectivity channel between stakeholders.</p>
                  <p className="m-0">translated designs to high quality & reusable react components.</p>
                </div>
              </div>
            </div>
  
            <div className="d-f fd-c" style={{ marginLeft: "40px" }}>
              <h1>recent work</h1>
  
              <div class="content mb-22">
                <a href="https://www.romacoutdoorspaces.ie" target="_blank">
                  <div class="content-overlay"></div>
                  <img class="content-image" src={romac} />
                  <div class="content-details fadeIn-bottom">
                    <h2 className="light">www.romacoutdoorspaces.ie</h2>
                  </div>
                </a>
              </div>
  
              <div class="content mb-32">
                <a href="https://www.mellio.com" target="_blank">
                  <div class="content-overlay"></div>
                  <img class="content-image" src={mellio} />
                  <div class="content-details fadeIn-bottom">
                    <h2 className="light">www.mellio.com</h2>
                  </div>
                </a>
              </div>
  
              <div class="content mb-32">
                <a href="https://wwww.cavancbt.ie" target="_blank">
                  <div class="content-overlay"></div>
                  <img class="content-image" src={cavancbt} />
                  <div class="content-details fadeIn-bottom">
                    <h2 className="light">www.cavancbt.ie</h2>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
    )
}

export default Experience;