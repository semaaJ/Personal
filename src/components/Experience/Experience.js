import React, { useState } from 'react';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import './Experience.css';

function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`vertical-tabpanel-${index}`}
      aria-labelledby={`vertical-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box p={3}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

const Experience = () => {
  const [value, setValue] = React.useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  return (
        <div className="root">
          <Tabs
            orientation="vertical"
            value={value}
            onChange={handleChange}
            className="tabs"
          >
            <Tab id="Shutterstock" label="Shutterstock" />
            <Tab id="DCU" label="Dublin City University" />
          </Tabs>

          <TabPanel value={value} index={0}>
            <div className="tabContentContainer">
              <div className="title">Software Engineer Intern @ <a className="employerLink" href="http://shutterstock.com/">Shutterstock</a></div>
              <div className="date">Apr 1 - 13 Sep 2019</div>
              <ul className="jobDetails">
                <li>
                  Worked, using agile & scrum methods, alongside a team developing an application which would provide a critical connectivity channel between stakeholders
                </li>
                <li>
                  Translated designs to high quality & reusable React components
                </li>
                <li>
                  Collaborated with editors to gather feedback for iterative design
                </li>
                <li>
                  Deployed code which met company standards of 85% test coverage (via Jest) & participated in peer code reviews for adherence to coding standards
                </li>
                <li>
                  Experienced developing back end services (Typescript)
                </li>
                <li>
                  Participated in a company hackathon
                </li>
                <li>
                  Utilised Javascript ES6, React, Redux, Node.js, Docker, RESTful APIs & Git
                </li>
              </ul>
            </div>
          </TabPanel>

          <TabPanel value={value} index={1}>
            <div className="tabContentContainer">
              <div className="title">
                Programming Tutor @ <a className="employerLink" href="https://www.computing.dcu.ie/event/computety">ComputeTY</a></div>
                Jan 22nd - 28th 2018
              <ul className="jobDetails">
                <li>
                  Selected for DCU's ComputeTY program which invites Transition Year students to attend a 3 week course introducing them to computer programming and encouraging them to study computer science
                </li>
                <li>
                  Worked alongside the students creating basic Android applications and games via Scratch.
                </li>
                <li>
                  Clarified, resolved and educated students on any issues encountered and the reasons behind these issues
                </li>
              </ul>
            </div>
          </TabPanel>    
        </div>
  );
}

export default Experience;