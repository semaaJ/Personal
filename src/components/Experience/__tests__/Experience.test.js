import React from 'react';
import Experience from '../Experience';
import { mount } from 'enzyme';
import '../../../__mocks__/intersectionObserverMock';
import "../../../setupTests"

test('should render Experience correctly', () => {
    const component = mount(<Experience />);
    expect(component.html()).toMatchSnapshot();
});

test('should change tab on tab click', () => {
    const component = mount(<Experience />);
    const tabButton = component.find('#DCU').first();
    tabButton.simulate('click');
    const tabTitle = component.find('.employerLink');
    expect(tabTitle.html()).toContain("ComputeTY");
});