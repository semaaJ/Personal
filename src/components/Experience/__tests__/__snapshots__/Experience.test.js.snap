// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Experience correctly 1`] = `"<div class=\\"root\\"><div class=\\"MuiTabs-root tabs MuiTabs-vertical\\"><div class=\\"MuiTabs-scroller MuiTabs-fixed\\" style=\\"overflow: hidden;\\"><div class=\\"MuiTabs-flexContainer MuiTabs-flexContainerVertical\\" role=\\"tablist\\"><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"true\\" id=\\"Shutterstock\\"><span class=\\"MuiTab-wrapper\\">Shutterstock</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" tabindex=\\"-1\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\" id=\\"DCU\\"><span class=\\"MuiTab-wrapper\\">Dublin City University</span><span class=\\"MuiTouchRipple-root\\"></span></button></div><span class=\\"PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator PrivateTabIndicator-vertical-4\\" style=\\"top: 0px; height: 0px;\\"></span></div></div><div role=\\"tabpanel\\" id=\\"vertical-tabpanel-0\\" aria-labelledby=\\"vertical-tab-0\\"><div class=\\"MuiBox-root MuiBox-root-5\\"><p class=\\"MuiTypography-root MuiTypography-body1\\"><div class=\\"tabContentContainer\\"><div class=\\"title\\">Software Engineer Intern @ <a class=\\"employerLink\\" href=\\"http://shutterstock.com/\\">Shutterstock</a></div><div class=\\"date\\">Apr 1 - 13 Sep 2019</div><ul class=\\"jobDetails\\"><li>Worked, using agile &amp; scrum methods, alongside a team developing an application which would provide a critical connectivity channel between stakeholders</li><li>Translated designs to high quality &amp; reusable React components</li><li>Collaborated with editors to gather feedback for iterative design</li><li>Deployed code which met company standards of 85% test coverage (via Jest) &amp; participated in peer code reviews for adherence to coding standards</li><li>Experienced developing back end services (Typescript)</li><li>Participated in a company hackathon</li><li>Utilised Javascript ES6, React, Redux, Node.js, Docker, RESTful APIs &amp; Git</li></ul></div></p></div></div><div role=\\"tabpanel\\" hidden=\\"\\" id=\\"vertical-tabpanel-1\\" aria-labelledby=\\"vertical-tab-1\\"></div></div>"`;
